documentation/src/main/asciidoc/topics/ref_hotrod_data_format.adoc

```diff
@@ -31,15 +28,14 @@ remoteCache.withDataFormat(rawKeyAndValues);
 
 [IMPORTANT]
 ====
-Using different marshallers and formats for keys, with `keyMarshaller()` and
-`keyType()` methods might interfere with the client intelligence routing
-mechanism and cause extra hops within the {brandname} cluster to perform the
-operation. If performance is critical, you should use keys in the format stored
-by the server.
+Using different marshallers and data formats for keys with `keyMarshaller()` and `keyType()` methods can interfere with client intelligence routing mechanisms, causing extra network hops within the {brandname} cluster.
+If performance is critical, you should use the format for keys in the remote cache.
 ====
 
 .Returning XML Values
 
+The following code snippet returns values as XML:
```

Comment:
I am confused by our headings here, as "Returning XML values" seems like a specific example that should be under the heading "Reading data in different encodings".

WDYT to making it so that we provide sections with common conversions, something like:

4. Data Conversion
4.3 Hot Rod Data Format API
Read and write operations on remote caches via the Hot Rod endpoint use the client marshaller by default. Hot Rod provides a DataFormat API for Java clients that you can use to perform cache operations with different media type encodings.
     // Different marshallers for key and values
     // Alternatively, use a custom value marshaller ...
     // Explain how it can be done
     
     4.3.1 XML
     // Concrete example with code sample and output
     4.3.2 JSON
     4.3.3 YAML

Reply:
I think this is a good idea but maybe too much for the purposes of the current PR. I've also said to Gustavo that this content is still something we'll need to expand for 13. I think the whole section on transcoding is more for power users so let's handle that in a future doc effort. I will make a note of this and factor it into the content plan.